File: fetch_test.py
import urllib.request
import json
import ssl

# Avoid SSL verification issues if any
ssl_context = ssl._create_unverified_context()

def check_url(url):
    try:
        req = urllib.request.Request(url, headers={'User-Agent': 'Mozilla/5.0'})
        with urllib.request.urlopen(req, context=ssl_context) as response:
            print(f"Success: {url} -> status {response.status}")
            # Try to read a bit
            data = json.loads(response.read().decode('utf-8'))
            print(f"Data type: {type(data)}")
            if isinstance(data, list):
                print(f"List length: {len(data)}")
                print(f"First element key/preview: {str(data[0])[:200]}")
            elif isinstance(data, dict):
                print(f"Dict keys: {list(data.keys())[:10]}")
                # Print some detail
                for k in list(data.keys())[:2]:
                    print(f"Key {k}: {str(data[k])[:100]}")
            return data
    except Exception as e:
        print(f"Failed to fetch {url}: {e}")
        return None

print("Checking Korean Bible...")
kor_data = check_url("https://raw.githubusercontent.com/thiagobodruk/bible/master/json/ko_ko.json")

print("\nChecking NIV Genesis Bible...")
# Let's try master first, then main if it fails
niv_data = check_url("https://raw.githubusercontent.com/aruljohn/Bible-niv/master/json/genesis.json")
if not niv_data:
    niv_data = check_url("https://raw.githubusercontent.com/aruljohn/Bible-niv/main/json/genesis.json")

File: fetch_test.js
const https = require('https');

function fetchJson(url) {
    return new Promise((resolve, reject) => {
        const options = {
            headers: {
                'User-Agent': 'Mozilla/5.0'
            }
        };
        https.get(url, options, (res) => {
            if (res.statusCode !== 200) {
                reject(new Error(`Failed to fetch ${url}, status: ${res.statusCode}`));
                return;
            }
            let body = '';
            res.on('data', (chunk) => body += chunk);
            res.on('end', () => {
                try {
                    const cleanBody = body.replace(/^\uFEFF/, '');
                    resolve(JSON.parse(cleanBody));
                } catch (e) {
                    reject(e);
                }
            });
        }).on('error', reject);
    });
}

async function main() {
    try {
        const niv = await fetchJson("https://raw.githubusercontent.com/aruljohn/Bible-niv/main/Genesis.json");
        const ch0 = niv.chapters["0"];
        console.log("ch0 keys:", Object.keys(ch0));
        console.log("ch0.chapter:", ch0.chapter);
        console.log("Type of ch0.verses:", typeof ch0.verses);
        if (Array.isArray(ch0.verses)) {
            console.log("ch0.verses is array, length:", ch0.verses.length);
            console.log("ch0.verses[0]:", ch0.verses[0]);
        } else {
            console.log("ch0.verses is object, keys count:", Object.keys(ch0.verses).length);
            console.log("ch0.verses keys:", Object.keys(ch0.verses).slice(0, 5));
            console.log("ch0.verses['1']:", ch0.verses["1"]);
        }
    } catch (e) {
        console.error(e);
    }
}

main();
